### PR TITLE
Use enhanced location as origin for reroutes

### PR DESCRIPTION
**Pull request by Guardiola31337**
## Description

This PR replaces origin location for reroutes from raw to enhanced

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Raw location was a source of bad reroutes and snappings to the opposite side of the route

### Implementation

Pass `tripSession.getEnhancedLocation()` as `origin` for reroutes

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

cc @mskurydin @SiarheiFedartsou @zugaldia @LukasPaczos

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Could you add a unit test @Guardiola31337?

**Comment by Guardiola31337** [comment]
@LukasPaczos Unit test added, this is ready for another round of 👀

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt [hunk 1]
```diff
@@ -225,6 +225,17 @@ class MapboxNavigationTest {
...
 
+    @Test
+    fun enhanced_location_used_for_reroute() {
+        val offRouteObserverSlot = slot<OffRouteObserver>()
+        verify { tripSession.registerOffRouteObserver(capture(offRouteObserverSlot)) }
+
+        offRouteObserverSlot.captured.onOffRouteStateChanged(true)
+
+        verify(exactly = 1) { tripSession.getEnhancedLocation() }
+        verify(exactly = 0) { tripSession.getRawLocation() }
```
Comment (LukasPaczos) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt]: An ideal test would be to mock `adjustedRouteOptionsProvider` and verify that the exact location returned by `tripSession.getEnhancedLocation` is pushed to `adjustedRouteOptionsProvider.getRouteOptions(location)`. Not going to block, but just something to consider.

Comment (Guardiola31337) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt]: Test added. Could you take a quick 👀 when you have a chance @LukasPaczos

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt [hunk 2]
```diff
@@ -225,6 +224,38 @@ class MapboxNavigationTest {
         verify(exactly = 1) { directionsSession.requestRoutes(any(), any()) }
     }
 
+    // TODO Fix test not working because of MapboxNavigationTelemetry calling functions are now suspendable
+    @Ignore
+    @Test
+    fun getEnhancedLocation_reRoute() {
```
Comment (kmadsen) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt]: These should be in a AdjustedRouteOptionsProviderTest

**Pull request merged**